files/usr/bin/cinnamon-xlet-makepot

```diff
@@ -0,0 +1,274 @@
+
+    if append:
+        pot_file.save(pot_path)
+    else:
+        pot_file.save(fpath=pot_path)
+
+def extract_settings_strings(data, dir, pot_file, parent=""):
+    for key in data.keys():
+        if key in ("description", "tooltip", "units", "title"):
+            comment = "%s->%s->%s" % (dir, parent, key)
+            save_entry(data[key], comment, pot_file)
+        elif key in "options":
+            opt_data = data[key]
+            for option in opt_data.keys():
+                if opt_data[option] == "custom":
+                    continue
+                comment = "%s->%s->%s" % (dir, parent, key)
+                save_entry(option, comment, pot_file)
+        elif key == "columns":
+            columns = data[key]
+            for i, col in enumerate(columns):
+                for col_key in col:
+                    if col_key in ("title", "units"):
+                        comment = "%s->%s->columns->%s" % (dir, parent, col_key)
+                        save_entry(col[col_key], comment, pot_file)
+        try:
+            extract_settings_strings(data[key], dir, pot_file, key)
+        except AttributeError:
+            pass
+
+def extract_metadata_strings(data, dir, pot_file):
+    for key in data:
+        if key in ("name", "description", "comments"):
+            comment = "%s->%s" % (dir, key)
+            save_entry(data[key], comment, pot_file)
+        elif key == "contributors":
+            comment = "%s->%s" % (dir, key)
+
+            values = data[key]
+            if isinstance(values, str):
+                values = values.split(",")
+
+            for value in values:
+                save_entry(value.strip(), comment, pot_file)
+
+def save_entry(msgid, comment, pot_file):
+    if not msgid.strip():
+        return
+
+    entry = pot_file.find(msgid)
+    if entry:
+        if comment not in entry.comment:
+            if entry.comment:
+                entry.comment += "\n"
+            entry.comment += comment
+    else:
+        entry = polib.POEntry(msgid=msgid, comment=comment)
+        pot_file.append(entry)
+
+def remove_empty_folders(path):
+    if not os.path.isdir(path):
+        return
+
+    # remove empty subfolders
+    files = os.listdir(path)
+    if len(files):
+        for f in files:
+            fullpath = os.path.join(path, f)
+            if os.path.isdir(fullpath):
+                remove_empty_folders(fullpath)
+
+    # if folder empty, delete it
+    files = os.listdir(path)
+    if len(files) == 0:
+        print('Removing empty folder:', path)
+        os.rmdir(path)
+
+def do_install(uuid, dir):
+    podir = os.path.join(dir, "po")
+    files_installed = 0
+    for root, subFolders, files in os.walk(podir):
+        for file in files:
+            locale_name, ext = os.path.splitext(file)
+            if ext == '.po':
+                lang_locale_dir = os.path.join(LOCALE_DIR, locale_name, 'LC_MESSAGES')
+                GLib.mkdir_with_parents(lang_locale_dir, 755)
```

Comment:
With the move to Python 3, maybe it isn't needed to use `GLib.mkdir_with_parents` and just use `os.makedirs(this_locale_dir, exist_ok=True)` here? One less module to import (**GLib**).

Furthermore, I think that the mode `755` passed to `GLib.mkdir_with_parents` is wrong. Since it's supposed to create folders with **read/write/exec** permissions for the current user, but instead it creates folders with only **write/exec** permissions for the current user. Using the mode `0o755` will create folders with the right permissions.

Reply:
hmm... yeah, that would simplify things a bit. I didn't realize that option had been added to python 3. That could also be used in cinnamon-settings to simplify things in a couple of places. I think I'll add it to my python 3 branch.

Reply:
Yes, I remember seeing the use of `mkdir` (the GNU coreutils command) in one of the other Python files. I found it weird, knowing that the `os.makedirs` method also exists in Python 2, although without the `exist_ok` parameter.

I didn't mention this in my previous message. But, if it is important for the directories to be generated with a **755** mode, then the code would be `os.makedirs(this_locale_dir, mode=0o755, exist_ok=True)`. The default mode used by `os.makedirs` is `0o777`.

Reply:
Yeah, I saw that in the docs. I already fixed it locally. I'm going to do a bit more testing and then I'll push my changes.